assets/src/domain/eventEditor/services/context/EntityListContext/DatetimesListContext.tsx

```diff
@@ -1,14 +1,32 @@
-import React, { createContext } from 'react';
+import React, { createContext, useMemo } from 'react';
 
-import { EntityListContextProps, ContextProviderProps } from '../types';
+import type { EntityListContextProps, ContextProviderProps } from '../types';
 import { DatetimesFilterStateManager, useDatesListFilterStateManager } from '../../filterState';
+import { useFilteredEntities } from '@appLayout/entityList';
+import { domain, datesList } from '@edtrServices/constants';
+import { useDatetimes } from '@edtrServices/apollo/queries';
+import type { Datetime } from '@edtrServices/apollo/types';
+import { getCacheIds } from '@sharedServices/predicates';
 
-export type DatetimesListContextProps = EntityListContextProps<DatetimesFilterStateManager>;
+export type DatetimesListContextProps = EntityListContextProps<DatetimesFilterStateManager, Datetime>;
 
 export const DatetimesListContext = createContext<DatetimesListContextProps>(null);
 
 export const DatetimesListProvider: React.FC<ContextProviderProps> = ({ children }) => {
+	const datetimes = useDatetimes();
 	const filters = useDatesListFilterStateManager();
-	const value = { filters };
+
+	const filtersStr = JSON.stringify(filters);
+	// memoize filter state
+	const filterState = useMemo(() => filters, [filtersStr]);
+
+	const filteredEntitiesList = useFilteredEntities(domain, datesList, datetimes, filterState);
+
+	const cacheIds = JSON.stringify(getCacheIds(filteredEntitiesList));
+	// memoize filtered entities
+	const filteredEntities = useMemo(() => filteredEntitiesList, [cacheIds]);
```

Comment:
would be better to move some of this memoization into `useFilteredEntities()` because then it will apply to both the dates list, tickets list, and any other entity list we build in the future.

So for example this code:

```js
	const cacheIds = JSON.stringify(getCacheIds(filteredEntitiesList));
	// memoize filtered entities
	const filteredEntities = useMemo(() => filteredEntitiesList, [cacheIds]);
```

could go into into `useFilteredEntities()` and automatically return a memoized entity list, so the code here in `DatetimesListContext` would just be

```js
const filteredEntities = useFilteredEntities(domain, datesList, datetimes, filterState);
```

but it would still be better to memoize the filterstate here since it's not really used elsewhere and will likely always be accessed via this context provider.

Might also be handy to return the stringified entity list cacheIDs for use in other components so that they don't have to recreate that logic like:

```js
const { entityListCacheKey, filteredEntities } = useFilteredEntities(domain, datesList, datetimes, filterState);
```

Reply:
I agree with moving that to `useFilteredEntities`, but not with the last suggestion - SRP.